src/js/entities-service/form-responses.js

```diff
@@ -8,25 +7,32 @@ const Entity = BaseEntity.extend({
     'formResponses:model': 'getModel',
     'formResponses:collection': 'getCollection',
     'fetch:formResponses:model': 'fetchFormResponse',
-    'fetch:formResponses:latest': 'fetchLatestResponse',
+    'fetch:formResponses:byMe': 'fetchByMe',
+    'fetch:formResponses:byPatient': 'fetchSubmitedByPatient',
   },
   fetchFormResponse(id, options) {
     if (!id) return new Model();
 
     return this.fetchModel(id, options);
   },
-  fetchLatestResponse(filter) {
-    const data = reduce(filter, (filters, value, key) => {
-      if (!value) return filters;
-      filters.filter[key] = value;
-      return filters;
-    }, { filter: {} });
+  fetchOrEmpty(url, data) {
+    return this.fetchBy(url, { data }).then(response => response || new Model());
+  },
+  fetchByMe({ actionId, patientId, formId }) {
+    const filter = actionId ? { action: actionId } : { patient: patientId, form: formId };
+
+    return this.fetchOrEmpty('/api/clinicians/me/form-responses/latest', { filter });
+  },
+  fetchSubmitedByPatient({ patientId, actionId, flowId, formId, actionTags, submittedAt }) {
+    const filter = {
+      ...(actionId && { actions: actionId }),
```

Comment:
this pattern was a chatgpt recommendation.. I'm on the fence.. the other option looks pretty much like it was with the filter method... but this translates the keys at the same time which is nice.